spec/ics-2-consensus-requirements/README.md

```diff
@@ -0,0 +1,89 @@
+
+2. Blocks have at least one direct child
+There is at least one direct child for all blocks, meaning that the lightclient logic can proceed the blocks one by one even in the worst case. If it not satisfied then there can be a point where the lightclient stops and cannot proceed, which halts IBC connection unexpectedly. This also can be violated when the blocks are restarted out of the consensus, for example in Tendermint, (1/2 < validators < 2/3) forked out the chain. This also will be covered in **byzantine recovery strategies**.
+
+3. If a block verifies another block then it is a descendant of the block
+Lightclient should not verify packets which are not in its chain. If the verifier returns true for a block that is not a descendant, it simply means that there is an error in the lightclient logic.
+
+4. Block can have a state only if it is a valid transition from the parent's(see the next paragraph)
+Block cannot have a state which is not a transition from its parent. Since the state is only defined for IBC logic, this only means that each block are running IBC-compatible application logic(does not care about the other modules' failure, including the handlers for the packets).
+
+These requirements allow channels to work safely without concerning about double spending attack. 
+If a block is submitted to the chain(and optionally passed some challenge period for fraud proof), then it is assured that the packet is finalized so the application logic can process it.
+
+### Technical Specification
+
+In a real world, `Block`s do not have to contain the full state data; it can contain an aggregated data set identifier, for example a merkle root, where the actual data latter given can be proven that it is a member of that set. If we don't consider about the data availability problem, this partial blocks, which is be called `Header`s, still satisfies `Block`.
+
+Following functions exist over `Header`s.
+
+* `height : Header -> int`
+Returns the height of the block.
+
+* `verify : Header -> Header -> Maybe<LightClientProof> -> bool`
+Verifies the latter block is a descendant of the former block, as defined in **lightclient specification**. Can take an additional lightclient proof argument.
+
+* `connection : Header -> ChainID -> Maybe<MemberProof> -> Connection`
+Returns the connection for `ChainID` as defined in **connection semantics**. Can take an additional set member proof argument.
+
+* `channel : Header -> ChainID -> PortID -> Channel`
+Returns the channel for `(ChainID, PortID)` pair, as defined in **channel semantics**. Can take an additional set member proof argument.
+
+### Example Implementation
+
+An example blockchain `B` is run by a single operator. If a block is signed by the operator, then it is valid. `B` contains `KVStore`, which is a mapping from `[byte]` to  `[byte]`. 
```

Comment:
Good example - I find the pseudocode a bit confusing though, can we make it a bit more like the ICS 1 example?